### ISSUE TITLE
Feature Request: valid prefixes or suffixes

### ISSUE THREAD

**Issue by thirtytwobits** [streetsidesoftware/vscode-spell-checker#4138]
First: thanks so much for this software. It's excellent. Grade-A work! ❤️

I'm wondering if you could add a feature to specify valid prefixes. This is different from `myprefix.*` in that it would signal a given patten is a valid word only if it is it the first part of a **compound word**. The example that inspired this request is `py`. So something like `py[valid-prfix]` (not suggesting configuration syntax here). This should then mark anything like `pytest` or `pyflyingtoasters` as correct but would mark `pytset`, `pyfliingtoasters` or just `py` as incorrect (`py`: unless the user also included `py` as a stand-alone entry).

For suffixes, vice-versa (`[valid-suffix]my-suffix`)

Alternatively you could allow any custom word to be marked as a valid compound word. The problem with this is it requires the word itself to be valid. This could be problematic if you worked for Terran Eater Hoteliers and your coding standard required `TEH` as a prefix when writing c functions (camel-case) `tehFoodOrderInitialization();`

**Comment by Jason3S** [comment]
@thirtytwobits,

The spell checker supports something like this in your own dictionaries. It is not exactly what you ask.


`README.md`

```markdown
# Checking compound words

The spell checker can combine words found in the same dictionary using `*` and `+`.
- `*` - optional
- `+` - required

You can test it in a document using a `cspell` directive:

cspell:words pre+ *mid* +fix bluue* *reeed +midddle+

The following words should be allowed:

- `premidddlefix`
- `prereeed`
- `bluuereeed`
- `bluuefix`
- `bluue`
- `reeed`

But not
- `midddle` - it requires a prefix and a postfix.
- `prebluue`
 
```

**Comment by thirtytwobits** [comment]
Indeed. This does work. Sorry for the noise. You can close this request.
